packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -67,16 +68,18 @@ export class Span implements api.Span, ReadableSpan {
     spanName: string,
     spanContext: api.SpanContext,
     kind: api.SpanKind,
+    clock: AnchoredClock,
```

Comment:
Is the change to the Span constructor considered a breaking change?

Reply:
Span constructor is not public and should not be used according to the spec. We just don't have a good way to make it private in JS.

Reply:
If we want to make it truly backward compatible we can make clock an optional argument and add it to the end?

Reply:
> Span constructor is not public and should not be used according to the spec. We just don't have a good way to make it private in JS.

yeah, but it is "public" in JS anyways and we didn't explicitly tag its constructor as an "internal" method.

As the weakmap has been removed and the chaining of AnchoredClock is based on `span._clock`, I believe moving this parameter to the end as an optional one can be a straightforward change.

Reply:
Also it will become an optional property so we will need to decide what we fall back on. Date?

Reply:
I don't have a strong opinion on this. Particularly, I found that the clock needs to be assigned to its parent span if it does not exist: https://github.com/open-telemetry/opentelemetry-js/pull/3134/files#diff-5c85570f9cc180caf78518f438d2153fd1d5791f4f25c1624f31d5ce0df790a2R134 (However, how can we get a span without the clock? Mixed usage of different SDK?)

Please feel free to dismiss this suggestion if you find this approach is good enough.

Reply:
`popagator.extract` creates a `NonRecordingSpan` which is a pure API thing and has no timestamps (it's never exported).
Similar suppress spans are `NonRecordingSpan` and also non sampled spans.

But all of them may be used as parent spans.

Reply:
As it happens there was a user who is using the span constructor directly. I advised them this is not supported but since we have released it then we can't break it.

fwiw: their usecase was collecting telemetry data from a service without network access and converting it to OTel data

Reply:
I updated the PR to make it strictly backwards compatible